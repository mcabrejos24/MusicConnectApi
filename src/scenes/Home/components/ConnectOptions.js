import './../styles.scss';

export default function ConnectOptions(props) {
    const { description } = props;
    const { image } = props;
    const { imageAlt } = props;
    const { url } = props;

    return (
        <div className="Connect-Options rounded-lg h-60 md:h-48 lg:h-64 xl:h-80">
            <a href={url}>
                <img src={image} alt={imageAlt} className="connect-options-img rounded-lg w-full h-full"/>
                <p className="options-description text-gray-100 text-2xl md:text-xl lg:text-2xl xl:text-4xl">{description}</p>
            </a>
        </div>
    );
}
