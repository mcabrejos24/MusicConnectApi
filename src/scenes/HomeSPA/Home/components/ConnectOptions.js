import { NavLink } from 'react-router-dom';
import '../styles.scss';

export default function ConnectOptions(props) {
    const { description } = props;
    const { image } = props;
    const { imageAlt } = props;
    const { url } = props;

    return (
        <div className="Connect-Options rounded-lg h-60 md:h-48 lg:h-64 xl:h-80">
            <NavLink to={ url }>
                <img src={image} alt={imageAlt} className="connect-options-img rounded-lg w-full h-full"/>
                <p className="options-description text-gray-100 text-xl sm:text-2xl md:text-base lg:text-2xl xl:text-3xl">{description}</p>
            </NavLink>
        </div>
    );
}
