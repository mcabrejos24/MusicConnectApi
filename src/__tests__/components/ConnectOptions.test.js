import React from "react";
import { shallow } from "enzyme";
import ConnectOptions from '../../components/ConnectOptions';
import toJson from "enzyme-to-json";

function createTestProps (props) {
  return {
    // common props
    description: "Pair both playlists here",
    image: "fakeimage.com",
    imageAlt: "my image alt",
    url: "path",
    // allow to override common props
    ...props,
  }
}

describe("renders ConnectOptions without crashing", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConnectOptions />);
  })
  it("tests ConnectOptions itself", () => {
    wrapper;
  });
  it("test ConnectOptions snapshot no props", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("tests ConnectOptions component with shalloww", () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<ConnectOptions {...props}/>);
  })
  it("has correct path", () => {
    const href = wrapper.find("a").prop("href");
    expect(href).toEqual("path");
  });
  it("has correct img and img alt", () => {
    const imgSrc = wrapper.find("img").prop("src");
    const imgAlt = wrapper.find("img").prop("alt");
    expect(imgSrc).toEqual("fakeimage.com");
    expect(imgAlt).toEqual("my image alt");
  });
  it("has correct description", () => {
    const pValue = wrapper.find("p").text();
    expect(pValue).toEqual("Pair both playlists here");
  });
  it('matches snapshot with props passed', () => {  
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
