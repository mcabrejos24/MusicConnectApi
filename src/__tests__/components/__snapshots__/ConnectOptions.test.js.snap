// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ConnectOptions without crashing test ConnectOptions snapshot no props 1`] = `
<div
  className="Connect-Options rounded-lg h-60 md:h-48 lg:h-64 xl:h-80"
>
  <a>
    <img
      className="connect-options-img rounded-lg w-full h-full"
    />
    <p
      className="options-description text-gray-100 text-xl sm:text-2xl md:text-base lg:text-2xl xl:text-3xl"
    />
  </a>
</div>
`;

exports[`tests ConnectOptions component with shalloww matches snapshot with props passed 1`] = `
<div
  className="Connect-Options rounded-lg h-60 md:h-48 lg:h-64 xl:h-80"
>
  <a
    href="path"
  >
    <img
      alt="my image alt"
      className="connect-options-img rounded-lg w-full h-full"
      src="fakeimage.com"
    />
    <p
      className="options-description text-gray-100 text-xl sm:text-2xl md:text-base lg:text-2xl xl:text-3xl"
    >
      Pair both playlists here
    </p>
  </a>
</div>
`;

exports[`tests component rendering through props with component 1`] = `ShallowWrapper {}`;
